viewer/packages/rendering/src/pointcloud-rendering/PointCloudObjectAppearanceTexture.ts

```diff
@@ -51,19 +54,39 @@ export class PointCloudObjectAppearanceTexture {
     data.set(styleData, 4 * objectId);
   }
 
-  private setObjectCollectionStyle(styledObjectSet: StyledPointCloudObjectCollection): void {
+  private setObjectCollectionStyle(styledObjectSet: StyledPointCloudVolumeCollection): void {
     if (!this._annotationIdsToObjectId) {
       throw new Error('Annotation ID to Object ID map not initialized');
     }
 
-    for (const annotationId of styledObjectSet.objectCollection.getAnnotationIds()) {
-      const objectId = this._annotationIdsToObjectId.get(annotationId);
+    const objectCollection = styledObjectSet.objectCollection;
 
+    const applyStyle = (transformedObjectId: number | string) => {
+      const objectId = this._annotationIdsToObjectId?.get(transformedObjectId);
       if (objectId === undefined) {
-        throw new Error('Could not find corresponding object ID for annotation ID' + annotationId);
+        throw new Error('Could not find corresponding object ID for ' + transformedObjectId);
       }
-
       this.setObjectStyle(objectId, styledObjectSet.style);
+    };
+
+    if (isPointCloudObjectCollection(objectCollection)) {
+      for (const annotationId of objectCollection.getAnnotationIds()) {
+        let transformedObjectId: number | string;
+        if (typeof annotationId === 'object' && 'externalId' in annotationId) {
+          transformedObjectId = (annotationId as DMInstanceRef).externalId;
+        } else if (typeof annotationId === 'number') {
+          transformedObjectId = annotationId;
+        } else {
+          throw new Error('Invalid annotation ID type');
+        }
+        applyStyle(transformedObjectId);
+      }
+    } else if (isDMInstanceRefPointCloudObjectCollection(objectCollection)) {
+      for (const instanceRef of objectCollection.getDataModelInstanceRefs()) {
+        applyStyle(instanceRef.externalId);
```

Comment:
I think we should include both `externalId` and `space` in this key, just to be entirely safe